### PR TITLE
rfc-support

### PR DESCRIPTION
**Pull request by dominikvoda**

### CODE REVIEW

**Review by tatyalien — APPROVED** [review]

File: src/DateTimeFormatter.php [hunk 1]
```diff
@@ -30,6 +30,12 @@ public static function formatAsAtom(DateTimeInterface $dateTime): string
     }
 
 
+    public static function formatAsRfc(DateTimeInterface $dateTime): string
```
Comment (michal-tichy-84) [inline src/DateTimeFormatter.php:33]: RFC? It sounds too much general to me. Maybe `formatWithMiliseconds`? Or `formatAsAtomWithMiliseconds`?

Comment (davidpilny) [reply src/DateTimeFormatter.php:33]: Its true we also had iso :thinking:

Comment (michal-tichy-84) [reply src/DateTimeFormatter.php:33]: Yes, but ISO is, AFAIK, relatively common and known datetime format, but RFC is not any "standard", just some "ticket" or "trigger to change".

File: src/DateTimeFromString.php [hunk 2]
```diff
@@ -33,6 +36,23 @@ public static function createFromAtom(string $dateTimeAtomString): DateTimeImmut
     }
 
 
+    public static function createFromRfc3339(
+        string $dateTimeInRfcString,
+        bool $withMilliseconds = false
+    ): DateTimeImmutable {
+        if ($withMilliseconds) {
```
Comment (tatyalien) [inline src/DateTimeFromString.php:43]: maybe

        if ($withMilliseconds) {
            // https://bugs.php.net/bug.php?id=75577
            $rfcFormat = PHP_VERSION_ID >= 70300
                ? DateTime::RFC3339_EXTENDED
                : 'Y-m-d\TH:i:s.uP';

            self::create($rfcFormat, $dateTimeInRfcString);
        } 

        return self::create($DateTime::RFC3339, $dateTimeInRfcString);

Comment (michal-tichy-84) [reply src/DateTimeFromString.php:43]: Where is an advantage?

File: src/DateTimeFormatter.php [hunk 3]
```diff
@@ -24,12 +24,26 @@ public static function formatTimestamp(int $timestamp, string $format): string
...
+    /**
+     * @deprecated Use `formatAsRfc3339`
+     */
     public static function formatAsAtom(DateTimeInterface $dateTime): string
     {
         return self::format($dateTime, DateTime::ATOM);
     }
 
 
+    public static function formatAsRfc3339(DateTimeInterface $dateTime, bool $withMilliseconds = false): string
```
Comment (davidpilny) [inline src/DateTimeFormatter.php:36]: Budeme vědět co to je? :-D

Comment (davidpilny) [reply src/DateTimeFormatter.php:36]: To zní jako dítě Elona Muska

Comment (michal-tichy-84) [reply src/DateTimeFormatter.php:36]: No to nevim, to si musíš odpovědět sám :-P

File: src/DateTimeFromString.php (hunk 2 above)
Comment (tatyalien) [reply src/DateTimeFromString.php:43]: i hate else :)

Comment (davidpilny) [reply src/DateTimeFromString.php:43]: Its true else is kind of forbidden :-D Maybe you should have another private method for it.

File: src/DateTimeFromString.php [hunk 4]
```diff
@@ -34,12 +36,18 @@ public static function createFromAtom(string $dateTimeAtomString): DateTimeImmut
...
 
-    public static function createFromRfc(string $dateTimeInRfcString): DateTimeImmutable
-    {
-        // https://bugs.php.net/bug.php?id=75577
-        $rfcFormat = PHP_VERSION_ID >= 70300
-            ? DateTime::RFC3339_EXTENDED
-            : 'Y-m-d\TH:i:s.uP';
+    public static function createFromRfc3339(
+        string $dateTimeInRfcString,
+        bool $withMilliseconds = false
```
Comment (dominikvoda) [inline src/DateTimeFromString.php:41]: boolean flag arguments are dirty habit... not sure it we should have them here :/

Comment (michal-tichy-84) [reply src/DateTimeFromString.php:41]: Let's discuss it at the daily

File: src/DateTimeFromStringTest.php [hunk 5]
```diff
@@ -2,86 +2,183 @@
...
 namespace BrandEmbassy\DateTime;
 
+use BrandEmbassy\DateTime\Format\Rfc3339ExtendedFormat;
 use DateTime;
 use DateTimeZone;
 use PHPUnit\Framework\Assert;
 use PHPUnit\Framework\TestCase;
+use function sprintf;
 
 class DateTimeFromStringTest extends TestCase
```
Comment (janveselynet) [inline src/DateTimeFromStringTest.php:12]: final

**Review by janveselynet — APPROVED** [review]

File: src/DateTimeFromString.php [hunk 6]
```diff
@@ -24,19 +25,24 @@ public static function create(string $format, string $dateTimeString): DateTimeI
...
+
+        return self::createFromFormat($format, $dateTimeStringInRfc3339ExtendedString);
     }
 
 
     /**
      * @throws InvalidDateTimeStringException
      */
-    public static function createWithTimezone(
+    public static function createWithTimezoneAndFormat(
```
Comment (dominikvoda) [inline src/DateTimeFromString.php:45]: createWithTimezoneFromFormat

**Review by dominikvoda — COMMENTED** [review]
I think it's ok :D I hope actually :D

**Review by michal-tichy-84 — APPROVED** [review]
Approved also by D1

**Pull request merged**